linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/VarGen.scala

```diff
@@ -21,89 +21,130 @@ import org.scalajs.ir.Types._
 
 import org.scalajs.linker.backend.javascript.Trees._
 
-/** Manages name generation for non-local, generated fields. */
+/** Manages name generation for non-local, generated fields.
+ *
+ *  We distinguish three types of linker generated identifiers:
+ *
+ *  - classVar: Vars pertaining to a given class.
+ *  - coreJSLibVar: Vars defined by the CoreJSLib.
+ *  - fileLevelVar: Vars that are local to an individual file.
+ *
+ *  At the moment, this distinction is theoretical. It will become relevant for
+ *  module splitting (#2681).
+ */
 private[emitter] final class VarGen(nameGen: NameGen,
     mentionedDangerousGlobalRefs: Set[String]) {
 
   import nameGen._
 
-  def codegenVar(field: String, typeRef: NonArrayTypeRef)(
-      implicit pos: Position): VarRef = {
-    VarRef(codegenVarIdent(field, typeRef))
-  }
+  // ClassName scoped.
 
-  def codegenVar(field: String, className: ClassName)(implicit pos: Position): VarRef =
-    codegenVar(field, genName(className))
+  def classVar(field: String, className: ClassName)(implicit pos: Position): VarRef =
+    VarRef(genericIdent(field, genName(className)))
 
-  def codegenVar(field: String, className: ClassName, fieldName: FieldName)(
+  def classVar(field: String, className: ClassName, fieldName: FieldName)(
       implicit pos: Position): VarRef = {
-    codegenVar(field, className, fieldName, NoOriginalName)
+    classVar(field, className, fieldName, NoOriginalName)
   }
 
-  def codegenVar(field: String, className: ClassName, fieldName: FieldName,
+  // ClassName, FieldName scoped.
+
+  def classVar(field: String, className: ClassName, fieldName: FieldName,
       origName: OriginalName)(
       implicit pos: Position): VarRef = {
-    codegenVar(field, genName(className) + "__" + genName(fieldName), origName)
+    VarRef(genericIdent(field, genName(className) + "__" + genName(fieldName), origName))
   }
 
-  def codegenVar(field: String, className: ClassName, methodName: MethodName)(
+  // ClassName, MethodName scoped.
+
+  def classVar(field: String, className: ClassName, methodName: MethodName)(
       implicit pos: Position): VarRef = {
-    codegenVar(field, className, methodName, NoOriginalName)
+    classVar(field, className, methodName, NoOriginalName)
   }
 
-  def codegenVar(field: String, className: ClassName, methodName: MethodName,
+  def classVar(field: String, className: ClassName, methodName: MethodName,
       origName: OriginalName)(
       implicit pos: Position): VarRef = {
-    codegenVar(field, genName(className) + "__" + genName(methodName), origName)
+    VarRef(genericIdent(field, genName(className) + "__" + genName(methodName), origName))
   }
 
-  def codegenVar(field: String, subField: String,
-      origName: OriginalName = NoOriginalName)(
+  // TypeRef scoped.
+
+  def classVar(field: String, typeRef: NonArrayTypeRef)(
```

Comment:
Yes, maybe. If so, then we should probably call them `classVar`, `methodVar`, `fieldVar` as well?

Reply:
Not necessarily. `typeRefVar` is really special because sometimes it goes to `coreJSLibVar`, sometimes to `classVar`. Method-based and field-based don't have that problem. They're always `classVar`s at the end of the day.

Reply:
Fair enough :)